nucleus/common/glassfish-api/src/main/java/org/glassfish/api/invocation/InvocationManagerImpl.java

```diff
@@ -173,7 +177,11 @@ public void setThreadInheritableInvocation(List<? extends ComponentInvocation> p
             throw new InvocationException();
         }
 
-        iter.next(); // the last is the current is "invocation"
+        ComponentInvocation current = iter.next(); // the last is the current is "invocation"
+        if (invocation != current) {
+            LOGGER.log(WARNING, "postInvoke not called with top of the invocation stack. Expected {0} but was: {1}",
+                    new Object[] { current, invocation });
```

Comment:
Last Nitpick would be, that only `EjbInvocation` provides reasonable `toString()`. It would be great to add one into `WebComponentInvocation` and `ComponentInvocation`.

Reply:
Good point, I remember checking it myself but then forgot to do something about it. Too many context switches. Will look into it tomorrow.